fragile/optimize/env.py

```diff
@@ -18,6 +18,7 @@ class Function(Environment):
 
     def __init__(
         self, function: Callable[[numpy.ndarray], numpy.ndarray], bounds: Bounds,
+            out_of_domain_check: Callable[[numpy.ndarray], bool] = None
```

Comment:
Maybe it's simpler if we give room to the end user to implement the function in a vectorized way (see comments below)
```suggestion
            out_of_domain_check: Callable[[numpy.ndarray], np.ndarray] = None
```

Reply:
done